### PR TITLE
[WIP] Fix elasticsearch container problem

### PR DESCRIPTION
**Pull request by luchiago**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

I wasn't able to communicate Elasticsearch with the other containers of dev to resulting in errors and problems with Faraday. So, I came up with this solution to make visible for all containers the volumes that we are using. But I still cannot solve #6817 

## Related Tickets & Documents
[#6923](https://github.com/thepracticaldev/dev.to/issues/6923#issuecomment-608013109)

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

docker

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -65,3 +65,10 @@ services:
         - ./docker_data/elasticsearch/data:/usr/share/elasticsearch/data
     ports:
       - "9200:9200"
+
+volumes:
```
Comment (citizen428) [inline docker-compose.yml:69]: I understand the URL change, but why is this necessary? Most of the services don't have volumes and even the ones who do shouldn't need to share them, which is what this option is for.

File: docker-compose.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ services:
       RAILS_ENV: development
       DATABASE_URL: postgresql://devto:devto@db:5432/PracticalDeveloper_development
       YARN_INTEGRITY_ENABLED: "false"
-      ELASTICSEARCH_URL: http://elasticsearch:9200
+      ELASTICSEARCH_URL: elasticsearch
```
Comment (citizen428) [inline docker-compose.yml:15]: Does this actually work for you? The URL doesn't specify the protocol or port, which seems odd.

Comment (citizen428) [reply docker-compose.yml:15]: To clarify: `elasticearch` only replace the IP/domain part of the URL, but we still need to specify the port etc.

File: docker-compose.yml (hunk 1 above)
Comment (luchiago) [reply docker-compose.yml:69]: I thought it's necessary to provide access to elasticsearch by the others containers and fix this problem
```shell
org.elasticsearch.bootstrap.StartupException: ElasticsearchException[failed to bind service]; nested: AccessDeniedException[/usr/share/elasticsearch/data/nodes
```

File: docker-compose.yml (hunk 2 above)
Comment (luchiago) [reply docker-compose.yml:15]: It worked, but I didn't know that we need to explicitly specify the port and protocol :sweat_smile:

File: docker-compose.yml (hunk 1 above)
Comment (citizen428) [reply docker-compose.yml:69]: The other containers don't need access to Elasticsearch's volume, as they are not writing to it.

**Comment by luchiago** [comment]
@citizen428 I think #7378 will solve permissions problems with elasticsearch and solve #6923. Should I keep investigating with this PR or close?

**Comment by citizen428** [comment]
@luchiago Let's please close this for now, we can always reopen it at a later date. And thanks for investigating this/spending time on it, much appreciated.
